rcl_yaml_param_parser/src/parser.c

```diff
@@ -1345,7 +1345,9 @@ static rcutils_ret_t parse_key(
             break;
           }
 
-          ret = rem_name_from_ns(ns_tracker, NS_TYPE_NODE, allocator);
+          allocator.deallocate(node_name_ns, allocator.state);
```

Comment:
Isn't there another leak up above if `find_node` returns something other than `RCUTILS_RET_OK`?

Reply:
How about `deallocate` right after find_node?

```
ret = find_node(node_name_ns, params_st, node_idx);
allocator.deallocate(node_name_ns, allocator.state);
if (RCUTILS_RET_OK != ret) {
  break;
}
```

Reply:
Yeah, that should fix it.